### PR TITLE
support for touch events on mobile

### LINKED ISSUE #14: Doesn't work on mobile
Tried both using chrome dev tools to emulate a mobile environment as well as using the sandbox demo url from my android phone also using chrome.  Dragging w/touch doesn't work.

### PR DESCRIPTION
**Pull request by arminfro**
I've made some changes to support touch events on mobile. 

If you like the changes I'd be happy to see them merged.

This would close #14

### CODE REVIEW

**Comment by mlejva** [comment]
Hey, thank you for the PR! This looks good from the first glance. I'll try it out today.

I'm curious though can you elaborate on what's the thinking behind the `condition` parameter [here](https://github.com/devbookhq/splitter/blob/ce8e124468584e86655971ac275d9c4ecd0aa299/src/useEventListener.ts#L5)?
Thanks!

**Comment by mlejva** [comment]
Hey, I just tried the your touch supported version in the [example project](https://github.com/devbookhq/splitter/tree/master/example) and I'm getting some weird behavior. 

Maybe I'm doing something wrong? If so, can you provide a CodeSandbox or similar example?

[Here's](https://imgur.com/a/10v8u9X) the video of me trying to use the splitter on my phone

File: src/index.tsx [hunk 1]
```diff
@@ -297,24 +303,33 @@ function Split({
     adjustSize(direction, offset);
   }, [state.isDragging, state.draggingIdx, state.pairs, adjustSize]);
 
-  function handleGutterMouseDown(gutterIdx: number, e: MouseEvent) {
-    e.preventDefault();
```
Comment (arminfro) [inline src/index.tsx:301]: I've got an error message  while dragging with touch caused by this line:
`Unable to preventDefault inside passive event listener invocation`

I haven't found a use case for this. Everything works without this call, is there any specific reason to call `preventDefault`?
If yes, I'm happy to revert and migrate this line :)

**Comment by arminfro** [comment]
Thanks for trying out. I should have tested the example project with my changes. 

The issue was the default scroll behaviour of the `touchmove` event. This is now handled with [`preventDefault` in case of a touch event](https://github.com/devbookhq/splitter/blob/791d82e/src/index.tsx#L323)

Now I've tested the example project with firefox and chrome on mobile and on desktop.

**Comment by mlejva** [comment]
Just tested it and all looks good. Thank you for the PR! I'll make a new release now.

**Pull request merged**